### DISCUSSION TITLE
Sonoff TX T2EU

### DISCUSSION THREAD

**Discussion by AndreScalaPT** [arendst/Tasmota#11090] [Q&A] [answered]
Hi everyone,

I'm trying to find how can I flash my Sonoff TX T2EU devices.

Where can I found what connects to what to be able to flash it?
I search everyone and I can't find any information abou the EU versions!

Thanks very much

**Comment by barbudor** [comment, marked as answer]
Serial flashing seems to be the way to go
There's are quite a few guide on the Internet that explain how to access the RX, TX pins. GPIO0 being the Button1 can be lower to GND by pressing it at reset/power-up

**Reply by AndreScalaPT** [reply]
Can you point me one or two?
I found 2 or 3 but the comments weren't positives to a point they break them lol
Thanks

**Reply by barbudor** [reply]
Most of them shows photos where the pins are clearly identified on the PCB
Do you have photos of yours ?

**Reply by AndreScalaPT** [reply]
![A](https://user-images.githubusercontent.com/65869347/108774230-b0979880-755f-11eb-9f1b-fabfd4b32723.jpg)
![B](https://user-images.githubusercontent.com/65869347/108774233-b2615c00-755f-11eb-8479-1d2ebd717f7f.jpg)

**Reply by AndreScalaPT** [reply]
I've connected:

GND - GND
TX - RX
RX - TX
VCC - VCC

This information is the same most the time in all guides that I saw, but the tasmotaizer don't find nothing connected

**Reply by barbudor** [reply]
And GPIO0 is on Button1 so you should connect VCC while pressing on Button1 to force the ESP in flashing mode and then start flashing

What did you tried for now ?
What is you USB Serial adapter ? Does it provide a good and reliable 3.3V power ?
Only adapter with a dedicated 3.3V regulator will work, such as the "Golden CH340G" recommended model

**Reply by AndreScalaPT** [reply]
So, I connect like I'm doing right?

GND - GND
TX - RX
RX - TX
VCC - VCC

Then before connecting the USB I keep pressing the white button? And the wireless light should not blink!

I bought the FTDI Adapter FT232RL USB to TTL Serial for 3.3V and 5V

![aaa](https://user-images.githubusercontent.com/65869347/108775621-9eb6f500-7561-11eb-8b8d-8e5c69b0d913.jpg)

**Reply by AndreScalaPT** [reply]
I connected like I said before.

Then I tried connect the USB while pressing the white button;
Then I tried connect the USB while pressing the button and connect a cable to GRD and touch the second leg in that big black chip

The pc don't recognize the board

**Reply by barbudor** [reply]
This is generally a bad adapter because they can't provide enough power on 3V3 as ESP based device need generally more than 200mA to work correctly.

Alternatively if you have another good source of 3V3 power supply.

If you have a NodeMCU, you can also use it as a USD/serial adapter by maintaining the on-board ESP module in RESET

**Reply by barbudor** [reply]
Have to go for today but you could join the Tasmota [Discord](https://discord.gg/Ks2Kzd4) server, in channel #flashing-issues to get more interactive help

**Reply by AndreScalaPT** [reply]
Thanks for your help, I'll do :D
Best regards

**Comment by wexivi** [comment]
I have flashed several sonoff devices with serial connection and now tried to flash T2EU3C.
I succeeded to flash this with serial connection and connecting GND and TP2/GPIO0 on the other side. However I succeeded to connect it after flashing (tasmota-xxx wifi and ipaddress192.168.4.1). However after defining my own wifi I am not able to connect it. Any ideas how to continue?
